service/post.service.js

```diff
@@ -127,3 +128,8 @@ exports.getPostsByUsername = async (username) => {
   ]).exec();
   return posts;
 };
+
+exports.extractUrlMetadata = async (url) => {
+  const metadata = await getLinkPreview(url);
```

Comment:
Awesome!
Can we get all the necessary data using this? Can you please provide the data sample we can get?

Reply:
https://github.com/ospfranco/link-preview-js#readme. the response sections mentions the data sample for different link types.

```
{
  url: "https://www.youtube.com/watch?v=MejbOFk7H6c",
  title: "OK Go - Needing/Getting - Official Video - YouTube",
  siteName: "YouTube",
  description: "Buy the video on iTunes: https://itunes.apple.com/us/album/needing-getting-bundle-ep/id508124847 See more about the guitars at: http://www.gretschguitars.com...",
  images: ["https://i.ytimg.com/vi/MejbOFk7H6c/maxresdefault.jpg"],
  mediaType: "video.other",
  contentType: "text/html; charset=utf-8",
  videos: [],
  favicons:["https://www.youtube.com/yts/img/favicon_32-vflOogEID.png","https://www.youtube.com/yts/img/favicon_48-vflVjB_Qk.png","https://www.youtube.com/yts/img/favicon_96-vflW9Ec0w.png","https://www.youtube.com/yts/img/favicon_144-vfliLAfaB.png","https://s.ytimg.com/yts/img/favicon-vfl8qSV2F.ico"]
}
```